cli/src/diff_util.rs

```diff
@@ -1135,28 +1151,57 @@ pub fn show_diff_summary(
     formatter: &mut dyn Formatter,
     mut tree_diff: TreeDiffStream,
     path_converter: &RepoPathUiConverter,
+    copy_records: &CopyRecords,
+    to_tree: &MergedTree,
 ) -> io::Result<()> {
+    let copied_sources: HashSet<&RepoPath> = copy_records
+        .records()
+        .map(|record| record.source.as_ref())
+        .collect();
+
     async {
         while let Some(TreeDiffEntry {
-            source: _, // TODO handle copy tracking
-            target: repo_path,
+            source: before_path,
+            target: after_path,
             value: diff,
         }) = tree_diff.next().await
         {
             let (before, after) = diff.unwrap();
-            let ui_path = path_converter.format_file_path(&repo_path);
-            if before.is_present() && after.is_present() {
-                writeln!(formatter.labeled("modified"), "M {ui_path}")?;
-            } else if before.is_absent() {
-                writeln!(formatter.labeled("added"), "A {ui_path}")?;
+            let before_ui_path = path_converter.format_file_path(&before_path);
+            let after_ui_path = path_converter.format_file_path(&after_path);
+            if before_path != after_path {
+                if to_tree
+                    .path_value(&before_path)
+                    .map_or(false, |v| v.is_absent())
```

Comment:
nit: I would prefer `.unwrap().is_absent()` to `map_or(false, ...)` so it's easier to notice later that we should fix this line (like line 1169 above)

Reply:
Done